src/main/java/com/jcraft/jsch/juz/Compression.java

```diff
@@ -29,10 +29,22 @@
   private Inflater inflater;
   private byte[] tmpbuf=new byte[BUF_SIZE];
   private byte[] inflated_buf;
+  private Session session;
 
   public Compression(){
   }
 
+  private void logMessage(int level, String message) {
+    Logger logger = session == null ? JSch.getLogger() : session.getLogger();
+    logger.log(level, message);
```

Comment:
I think we should add a check to make sure `logger.isEnabled(level)` here before actually logging the message?

Reply:
This would also apply to the `jzlib` implementation of `Compression` as well.

Reply:
That wouldn't make much sense, because the common reason to do this is to avoid concatenating a log-message that isn't used later on if `level > currentLevel`. In `logMessage` that concatenation has already taken place and `logger.logMessage` is supposed to do this check itself, so doing this check before doesn't save anything.

Reply:
If a user has a custom Logger that has a specific logging level disabled, we shouldn't emit a log message at said level.

Reply:
I've added a check and used a Supplier, so there is no need for all these `if(enabled)-logMessage`-cascades.